src/components/generics/vacation-test-forms/myVacationRequests.tsx

```diff
@@ -0,0 +1,274 @@
+
+interface Request {
+  id: number;
+  vacationType: string;
+  comment: string;
+  employee: string;
+  days: number;
+  startDate: string;
+  endDate: string;
+  status: string;
+}
+
+/**
+ * Renders vacation request table
+ */
+const renderVacationRequests = () => {
+  const classes = useEditorContentStyles();
+  const [dateFormat] = useState<string>("yyyy.MM.dd");
+  const [selectedVacationStartDate, setSelectedVacationStartDate] = useState<Date>(new Date());
+  const [selectedVacationEndDate, setSelectedVacationEndDate] = useState<Date>(new Date());
+  const [openRows, setOpenRows] = useState<boolean[]>([]);
+  const [newTextContent, setNewTextContent] = useState("");
+  const [newVacationType, setNewVacationType] = useState("");
+  const [datePickerView] = useState<CalendarPickerView>("day");
+  const [textContent] = useState("");
+  const [vacationType] = useState("");
+
+  /**
+  * Handle vacation type 
+  */
+  const handleVacationTypeChange = (event: SelectChangeEvent) => {
+    const contentValue = event.target.value;
+    setNewVacationType(contentValue as string);
+  };
+
+  /**
+  * Renders the vacation type selection
+  */
+  const renderVacationType = () => (
+    <FormControl
+      variant="standard"
+      sx={{
+        m: 1, minWidth: 165, marginBottom: 4
+      }}
+    >
+      <InputLabel>{ strings.editorContent.vacationType }</InputLabel>
+      <Select
+        value={newVacationType}
+        onChange={handleVacationTypeChange}
+        label={ strings.editorContent.vacationType }
+      >
+        <MenuItem value="Paid leave">{ strings.editorContent.paidLeave }</MenuItem>
+        <MenuItem value="Maternity leave">{ strings.editorContent.maternityLeave }</MenuItem>
+        <MenuItem value="Parental leave">{ strings.editorContent.parentalLeave }</MenuItem>
+        <MenuItem value="Unpaid leave">{ strings.editorContent.unpaidLeave }</MenuItem>
+        <MenuItem value="Surplus balance">{ strings.editorContent.surplusBalance }</MenuItem>
+      </Select>
+    </FormControl>
+  );
+
+  /**
+  * Handle vacation comment box content
+  */
+  const handleVacationCommentContent = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const contentValue = event.target.value;
+    setNewTextContent(contentValue);
+  };
+
+  /**
+  * Renders vacation comment box
+  */
+  const renderVacationCommentBox = () => (
+    <TextField
+      id="outlined-multiline-flexible"
+      multiline
+      maxRows={5}
+      label={ strings.editorContent.leaveAComment }
+      variant="outlined"
+      value={newTextContent}
+      onChange={handleVacationCommentContent}
+    />
+  );
+
+  /**
+* Handle vacation apply button
+*/
+  const handleVacationApplyButton = () => {
+    return (
+      // eslint-disable-next-line no-console
+      console.log(`this is START DATE ${selectedVacationStartDate} and this is END DATE${selectedVacationEndDate} and this is TEXT CONTENT ${textContent}. Vacation type ${vacationType}`)
```

Comment:
Is this console log needed or is this function to be implemented in the future, in which case a comment about not yet implemented would be a good thing